pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -461,6 +465,47 @@ async def publish_client(self, working_dir, from_release_tag, release_name, buil
         # Publish the clients to our S3 bucket.
         await exectools.cmd_assert_async(f"aws s3 sync --no-progress --exact-timestamps {base_to_mirror_dir}/{build_arch} s3://art-srv-enterprise/pub/openshift-v4/{build_arch}", stdout=sys.stderr)
 
+    def publish_baremetal_installer_binary(self, from_release_tag: str, client_mirror_dir: str):
+        release_pullspec = f'{constants.QUAY_RELEASE_REPO_URL}:{from_release_tag}'
+        image_stat, baremetal_installer_pullspec = get_release_image_pullspec(release_pullspec, 'baremetal-installer')
+
+        if image_stat:  # != 0, image does not exist
+            self._logger.error('Error getting baremetal-installer binary from %s', release_pullspec)
+            return
```

Comment:
`get_release_image_pullspec` already has `check=True` so no need to check the exist code.

Reply:
Right, but what we want to do here is exit from `publish_baremetal_installer_binary` function, because if `oc adm release info` fails, nothing else can be done. `check_status` in `common_oc_wrapper` will only log a warning

Reply:
If `oc adm release info` fails, an exception will be raised and the pipeline will abort unless you capture the exception.

Reply:
Fixed with https://github.com/openshift-eng/aos-cd-jobs/pull/3774/commits/113c9a074dbe1daf09aa37634fff6cdb72f5e5e4